modules/core/src/main/java/org/apache/ignite/internal/pagemem/wal/record/EncryptedRecord.java

```diff
@@ -43,7 +43,7 @@ public EncryptedRecord(int grpId, RecordType plainRecType) {
 
     /** {@inheritDoc} */
     @Override public RecordType type() {
-        return RecordType.ENCRYPTED_RECORD;
+        return RecordType.ENCRYPTED_RECORD_V2;
```

Comment:
Could you please describe in short why you need to change the type of this WAL record?
WAL record was not changed itself, so it looks weird to me.

Reply:
For binary compatibility.
For ENCRYPTED_RECORD_V2 we reading 1 extra byte (encryption key identifier).
(see change in RecordDataV1Serializer.readEncryptedData)

Reply:
> (see change in RecordDataV1Serializer.readEncryptedData)
Yep, I saw this, but it does not explain why you modify the existing WAL record.

In my understanding, WAL records must be immutable in order to avoid compatibility issues in the future.
I would prefer adding a new WAL record (it can extend an existing one).
Perhaps, RecordDataV1Serializer implementation will be cleaner in that case.

Reply:
Thanks for the suggestion, as far as I understand you are suggesting not to modify any existing WAL record classes.
Could you please give an example of a compatibility issue that we may face in the future (in theory)?

Reply:
Hello @xtern,

I think changing already existing WAL records is an error-prone approach.
Today we changed a type; tomorrow we will change/add a field and skip the corresponding changes in the serializer code.
It is just good practice not to change existing records, IMHO. Anyway, it is up to you.

Reply:
I updated this patch and kept the original WAL records (EncryptedRecord and MasterKeyChangeRecord) unchanged.